pkg/collector/corechecks/servicediscovery/envs/envs.go

```diff
@@ -35,17 +35,21 @@ type Variables struct {
 
 // NewVariables returns a new [Variables] to collect env. variables.
 func NewVariables(vars map[string]string) Variables {
+	if len(vars) > 0 {
+		// Only some tests pass a non-empty parameter. Check that input variables match the target value.
+		for env := range vars {
+			if _, ok := targets[env]; !ok {
+				return Variables{}
+			}
+		}
+	}
```

Comment:
1. iterating over an empty collection is not executing anything, so you can remove the first condition
2. The current code rejects any input that contains at least one unsupported environment variable, that's not the equivalent of the change. You should have an insertion filter to store only supported environment variables

Reply:
on 1, I removed length condition
on 2, only unit tests call this function with non-empty map, it is enough to have one non-targeted variable in input to fail the unit test. Regular envs reading sets input map empty and fills it reading /proc